### PR TITLE
Deprecated `tight_bbox` and `tight_layout` modules

### PR DESCRIPTION
**Pull request by oscargus**
## PR Summary

Related to #16181 

The modules `matplotlib.tight_bbox` and `matplotlib.tight_layout` are deprecated.

Will add release notes once the tests pass (and I have a PR-number).

## PR Checklist

**Tests and Styling**
- [N/A] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [N/A] New features are documented, with examples if plot related.
- [N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Review by timhoffm — COMMENTED** [review]
The doc build still fails with
```
/home/circleci/project/doc/api/prev_api_changes/api_changes_1.4.x.rst:152: WARNING: py:obj reference target not found: tight_bbox.adjust_bbox
/home/circleci/project/doc/api/prev_api_changes/api_changes_1.4.x.rst:152: WARNING: py:obj reference target not found: tight_bbox.process_figure_for_rasterizing
/home/circleci/project/doc/api/prev_api_changes/api_changes_2.2.0.rst:162: WARNING: py:obj reference target not found: tight_layout.auto_adjust_subplotpars
/home/circleci/project/doc/api/prev_api_changes/api_changes_3.0.1.rst:4: WARNING: py:obj reference target not found: tight_layout.auto_adjust_subplotpars
/home/circleci/project/doc/api/prev_api_changes/api_changes_3.0.1.rst:4: WARNING: py:obj reference target not found: tight_layout.get_tight_layout_figure
```
These historic API change notes reference the now non-public API. The way to handle this is to change the links to verbatim as that API is no longer public and available in the docs. E.g. for the first one, change

```
`.tight_bbox.adjust_bbox`
```
to
```
``matplotlib.tight_bbox.adjust_bbox``
```

**Comment by oscargus** [comment]
Thanks! I had that question in #22132, so I'll fix both.

**Comment by timhoffm** [comment]
I did a specific comment in #22133 as well.

**Comment by oscargus** [comment]
Thanks @timhoffm !

I've updated both and will set them to be ready when the tests pass.

I decided to not put links in the deprecation notices to not have to edit them when the modules actually are removed.

File: doc/api/next_api_changes/deprecations/22134-OG.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Modules ``tight_bbox``, ``tight_layout`` deprecated
+------------------------------------------------------------
```
Comment (timhoffm) [inline doc/api/next_api_changes/deprecations/22134-OG.rst:2]:
```suggestion
Modules ``tight_bbox`` and ``tight_layout``
-------------------------------------------
```

**Review by timhoffm — APPROVED** [review]

**Review by timhoffm — COMMENTED** [review]
It just occured to me that we can an should keep the history for git blame. If you create a new `_tight_bbox.py`, copy everything over and add the stub in `tight_bbox.py`, then all of `_tight_bbox.py` will be marked as your change and not have any history behind that.

To preserve the history:

1. `git mv tight_bbox.py _tight_bbox.py` and put this into a commit.
2. Create a new `tight_bbox.py` with the deprecation and put this into a second commit.

It's important that the move is in a separate commit. Otherwise git assumes you created `_tight_bbox.py` and modified `tight_bbox.py`.

Sorry to bring this in at a quite late review stage, but it's really important for us to be able to easily dig into the code history.

**Comment by oscargus** [comment]
Sure! Shouldn't be much work! (And clearly makes sense, although I didn't think about it before.)

File: doc/api/next_api_changes/deprecations/22134-OG.rst (hunk 1 above)
Comment (oscargus) [reply doc/api/next_api_changes/deprecations/22134-OG.rst:2]: Gah! That is what happens when you copy and paste too much and think too little... Thanks!

**Review by timhoffm — APPROVED** [review]

**Review by jklymak — APPROVED** [review]

**Pull request merged**